internal/util/regexp.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021-2023 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"fmt"
+	"regexp"
+	"sync"
+)
+
+type RegexpCache struct {
+	cache map[string]*regexp.Regexp
+	mu    *sync.Mutex
+}
+
+func NewRegexpCache() *RegexpCache {
+	return &RegexpCache{
+		cache: make(map[string]*regexp.Regexp),
+		mu:    &sync.Mutex{},
```

Comment:
The zero value of mutex is usable out of the box so it doesn't need to be initialised.

```suggestion
		
```